lib/packwerk/application_load_paths.rb

```diff
@@ -8,20 +8,15 @@ module ApplicationLoadPaths
     class << self
       extend T::Sig
 
-      sig { returns(T::Array[String]) }
-      def extract_relevant_paths
-        assert_application_booted
+      sig { params(root: String, environment: String).returns(T::Array[String]) }
+      def extract_relevant_paths(root, environment)
+        require_application(root, environment)
```

Comment:
I think the application technically also needs to be booted when extracting / comparing inflections ( `InflectionsFile` generator and `check_inflection_file` in `ApplicationValidator`).

I assume this works because things happen to be executed in the right order.